log4j-api-test/src/test/java/org/apache/logging/log4j/LoggerTest.java

```diff
@@ -203,11 +191,18 @@ public String get() {
             }
         });
         logger.traceExit(msg, 3);
-        assertEquals(2, results.size());
-        assertThat("Incorrect Entry", results.get(0), startsWith("ENTER[ FLOW ] TRACE Enter"));
-        assertThat("Missing entry data", results.get(0), containsString("doFoo(a=1, b=2)"));
-        assertThat("Incorrect Exit", results.get(1), startsWith("EXIT[ FLOW ] TRACE Exit"));
-        assertThat("Missing exit data", results.get(1), containsString("doFoo(a=1, b=2): 3"));
+        assertThat(results).hasSize(2);
+        assertThat(results.get(0)).startsWith("ENTER[ FLOW ] TRACE Enter").contains("doFoo(a=1, b=2)");
+        assertThat(results.get(1)).startsWith("EXIT[ FLOW ] TRACE Exit").contains("doFoo(a=1, b=2): 3");
+    }
+
+    @Test
+    public void flowTracingNoFormat() {
+        logger.traceEntry(null, 1, "2", new ObjectMessage(3));
+        logger.traceExit((String) null, 4);
+        assertThat(results).hasSize(2);
+        assertThat(results.get(0)).isEqualTo("ENTER[ FLOW ] TRACE Enter with params: [1, 2, 3]");
+        assertThat(results.get(1)).isEqualTo("EXIT[ FLOW ] TRACE Exit with: 3");
```

Comment:
The previous code printed "Enter param(1, 2, 3)", but used a format that depended on the number of parameters. The current code uses a `ParameterizedMessage("params: {}", (Object) params)`.